src/interface.rs

```diff
@@ -406,12 +534,47 @@ impl Interface {
             }
         }
 
+        let mut dhcp4_settings: Option<Dhcp4Settings> = None;
+        if let Some(ipv4_dhcp) = &self.ipv4_dhcp {
+            let mut dhcp_settings = Dhcp4Settings::default();
+            if let Some(hostname) = &ipv4_dhcp.hostname {
+                dhcp_settings.send_hostname = true;
+                dhcp_settings.hostname = Some(hostname.clone());
+            } else {
+                dhcp_settings.send_hostname = false;
+            }
+            dhcp_settings.send_release = Some(ipv4_dhcp.release_lease);
+            dhcp4_settings = Some(dhcp_settings);
+        }
+
+        let mut dhcp6_settings: Option<Dhcp6Settings> = None;
+        if let Some(ipv6_dhcp) = &self.ipv6_dhcp {
+            let mut dhcp_settings = Dhcp6Settings::default();
+            if let Some(hostname) = &ipv6_dhcp.hostname {
+                dhcp_settings.send_hostname = true;
+                // The problem here is if it's just the default hostname it shouldn't be hardcoded here...
```

Comment:
Exactly! From ifcft point it is `DHCLIENT6_HOSTNAME_OPTION=AUTO` which is default. But it get's translated to https://github.com/openSUSE/wicked/blob/master/client/suse/compat-suse.c#L6005 and then to xml here: https://github.com/openSUSE/wicked/blob/master/client/compat.c#L2478